web/core/hooks/use-reload-confirmation.tsx

```diff
@@ -1,41 +1,57 @@
-  //   (event: BeforeUnloadEvent) => {
-  //     if (!isActive || !showAlert) return;
-  //     event.preventDefault();
-  //     event.returnValue = "";
-  //   },
-  //   [isActive, showAlert]
-  // );
+  const handleBeforeUnload = useCallback(
+    (event: BeforeUnloadEvent) => {
+      if (!isActive || !showAlert) return;
+      event.preventDefault();
+      event.returnValue = "";
+    },
+    [isActive, showAlert]
+  );
 
-  // const handleRouteChangeStart = useCallback(
-  //   (url: string, { shallow }: { shallow: boolean }) => {
-  //     if (!isActive || !showAlert) return;
-  //     const leave = confirm("Are you sure you want to leave? Changes you made may not be saved.");
-  //     if (!leave) {
-  //       router.events.emit("routeChangeError", new Error("Route change cancelled by user"), url, shallow);
-  //       throw "routeChange aborted.";
-  //     }
-  //   },
-  //   [isActive, router.events, showAlert]
-  // );
+  const handleAnchorClick = useCallback(
+    (event: MouseEvent) => {
+      if (!isActive || !showAlert) return;
+      // Skip if event target is not available or defaultPrevented
+      if (!event.target || event.defaultPrevented) return;
+      // Skip control/command/option/alt+click
+      if (event.metaKey || event.ctrlKey || event.shiftKey || event.altKey) return;
+      // check if the event target is an anchor or a child of an anchor tag
+      const eventTarget = event.target as HTMLElement;
+      if (!eventTarget.closest("a")) return; // This is intentionally not type safe
+      // check if anchor target is _blank
+      const anchorElement = eventTarget.closest("a") as HTMLAnchorElement;
+      const isAnchorTargetBlank = anchorElement.getAttribute("target") === "_blank";
+      if (isAnchorTargetBlank) return;
+      // show confirm dialog
+      const leave = confirm("Are you sure you want to leave? Changes you made may not be saved.");
+      if (!leave) {
+        event.preventDefault();
+        event.stopPropagation();
+      }
+    },
+    [isActive, showAlert]
+  );
 
-  // useEffect(() => {
-  //   window.addEventListener("beforeunload", handleBeforeUnload);
-  //   router.events.on("routeChangeStart", handleRouteChangeStart);
+  useEffect(() => {
+    // handle browser refresh
+    window.addEventListener("beforeunload", handleBeforeUnload, true);
+    // handle anchor tag click
+    const body = document.querySelector("body");
+    if (!body) return;
+    body.addEventListener("click", handleAnchorClick, true);
+  // TODO: handle back / forward button click
 
-  //   return () => {
-  //     window.removeEventListener("beforeunload", handleBeforeUnload);
-  //     router.events.off("routeChangeStart", handleRouteChangeStart);
-  //   };
-  // }, [handleBeforeUnload, handleRouteChangeStart, router.events]);
+    return () => {
+      // cleanup
+      window.removeEventListener("beforeunload", handleBeforeUnload);
+      body.removeEventListener("click", handleAnchorClick, true);
```

Comment:
Can you try using window instead of body. It will not have difference but, window comes first and its recommended to attach to the closest node, and you can remove the check on line 45.

Reply:
Sure!